### PR TITLE
Replace deprecated .value with [()].

### LINKED ISSUE #5: Remove use of deprecated '.value'
Deprecation message:
`` H5pyDeprecationWarning: dataset.value has been deprecated. Use dataset[()] instead.``

### PR DESCRIPTION
**Pull request by greschd**
Fixes #5.

### CODE REVIEW

File: tests/test_save_load.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def test_file_method():
     """
     x = SimpleClass(6)
     with tempfile.NamedTemporaryFile() as named_file:
-        with h5py.File(named_file.name) as h5_file:
+        with h5py.File(named_file.name, 'r+') as h5_file:
```
Comment (mskoenz) [inline tests/test_save_load.py:58]: why this? did the default change?

**Comment by mskoenz** [comment]
why the "r+" change in open?
Rest looks fine

File: tests/test_save_load.py (hunk 1 above)
Comment (greschd) [reply tests/test_save_load.py:58]: Yeah, default will be 'r' in the future.

Comment (greschd) [reply tests/test_save_load.py:58]: At least, if the deprecation warning / message is to be believed..

**Comment by mskoenz** [comment]
I see, all good then 👍

**Comment by greschd** [comment]
Thanks for reviewing!

**Pull request merged**